# Geolocalización.csv

Contiene la información geográfica de cada inmueble, ubicación y sitios cercanos, además nos dice la planta del piso, siendo información de gran valor. El id de cada inmueble es su RK.

In [132]:
import pandas as  pd

In [133]:
df=pd.read_csv('../data/geolocalizacion.csv',sep=';')

In [134]:
df.dtypes

rk                   object
latitud             float64
longitud            float64
codigo_postal         int64
pais                  int64
provincia             int64
ciudad               object
distrito             object
barrio               object
calle                object
planta                int64
puerta               object
metro                object
cercanias            object
centro_comercial     object
escuela              object
parque               object
hospital             object
gimnasio             object
dtype: object

In [135]:
df.isna().sum()

rk                    0
latitud               2
longitud              2
codigo_postal         0
pais                  0
provincia             0
ciudad                0
distrito             98
barrio              357
calle                 0
planta                0
puerta               14
metro               154
cercanias           361
centro_comercial    166
escuela             161
parque              150
hospital            162
gimnasio            162
dtype: int64

In [136]:
(df.longitud.isna() | df.latitud.isna()).value_counts()

False    369
True       2
dtype: int64

Parece que de los inmuebles que no tenemmos los geo, no tenemos ninguno y en total son solo 2.

In [137]:
df.pais.value_counts()

1    371
Name: pais, dtype: int64

Parece que, por ahora sólo tenemos datos en este data set sobre España.



In [138]:
df.provincia.value_counts()

1    333
2     38
Name: provincia, dtype: int64

Sólo tenemos datos de dos provincias: 1-Madrid, 2-Barcelona. No aparecen NaN

In [139]:
df.distrito.isna().sum()

98

Tenemos 98 datos de DISTRITO en NAN

In [140]:
df.groupby(['provincia', 'ciudad', 'distrito'], dropna=False).count()['rk']

provincia  ciudad                    distrito                  
1          Alcalá De Henares         Alcala de Henares             1
                                     Juan de Austria               1
                                     Los Almendros                 1
                                     San Isidro - Los Almendros    1
           Alcorcón                  Centro                        2
                                                                  ..
2          Sant Adria De Besos       La Catalana                   1
                                     Sant Martí                    1
           Sant Joan Despi           Sant Joan Despí               1
           Santa Coloma De Gramanet  Can Mariné                    1
           Vallirana                 Centro                        1
Name: rk, Length: 108, dtype: int64

Para unificar estos datos y completarlos si fuera necesario podríamos usar reverse geocoding. Bien por coordenadas GPS, bien por calle y número 🔥🔥¡¡¡pero no tenemos número!!!

https://stackoverflow.com/questions/4013606/google-maps-how-to-get-country-state-province-region-city-given-a-lat-long-va

tb me recomienda geopy Nomatin

creo que clikalia usa google api. si lo usa podemos implementar mejor googleapi. controlar muy bien las llamadas que hacemos. ¿añadir un campo de última actualización de la info con info api?


In [141]:
df.puerta.value_counts()

A            73
B            57
C            48
D            41
2            18
1            16
3            14
Derecha      10
E             9
I             7
4             6
F             5
5             5
d             4
0             4
DCHA          3
H             3
G             2
Izquierda     2
ext           2
6             2
Exterior      1
303           1
IZQ           1
Ext.          1
420           1
B/Derecha     1
12            1
c             1
16            1
J             1
i             1
13            1
e             1
AD            1
Ático 4       1
27D           1
EC            1
463           1
M             1
izq           1
derecha       1
Izq           1
IZDA          1
a             1
Izda.         1
b             1
Name: puerta, dtype: int64

La columna de "puerta" parece bastante sucia. ¿Es siginificativo? En general NO pero parece que hay un dato que podemos extraer: 🔥ático

In [142]:
df.cercanias.value_counts()

Fuenlabrada Central                          2
La Serna, Fuenlabrada Central                1
La Avanzada-La Cueva, Fuenlabrada Central    1
Méndez Álvaro                                1
Torrejón de Ardoz                            1
Lavapiés-Embajadores Central                 1
Asamblea de Madrid Entrevías                 1
Zarzaquemada, Leganés Central                1
Este, Móstoles Central                       1
Name: cercanias, dtype: int64

aquí nos da el nombre del cercanías más cercano. A lo mejor sería buen idea encontrar distancia al cercanías más cercano. ¿recurrimos a api?

In [143]:
df.centro_comercial.value_counts()

Centro Comercial Plaza Río 2     15
Ventanal de la Sierra            11
Centro Comercial Alcalá Norte     9
El Corte Inglés                   8
Plaza Río 2                       8
                                 ..
H2O                               1
Centro Comercial Parque Sur       1
Maremagnum                        1
ABC Serrano                       1
Centro Comercial Arturo Soria     1
Name: centro_comercial, Length: 111, dtype: int64

In [144]:
df.centro_comercial.isna().sum()

166

In [145]:
df.centro_comercial.isnull().sum()

166

Tenemos 166 datos no válidos.

IDEM CON CENTRO COMERCIAL. aquí nos da el nombre del centro comercial más cercano. A lo mejor sería buen idea encontrar distancia al más cercano. ¿recurrimos a api?

In [146]:
df.escuela.value_counts()

Virgen de los Remedios                  11
Nuestra Señora del Pilar                 5
Santa Maria del Camino                   3
IES Tirso de Molina                      3
Colegio Reinado del Corazón de Jesús     3
                                        ..
Gabriel García Marquéz                   1
CEIP Giner de los Ríos                   1
IES Lope de Vega                         1
Tres Olivos                              1
Nazaret San Blas                         1
Name: escuela, Length: 172, dtype: int64

In [147]:
df.escuela.isnull().sum()

161

Tenemos 166 datos no válidos.

IDEM CON ESCUELAS. aquí nos da el nombre del escuela más cercano. A lo mejor sería buen idea encontrar distancia al más cercano. ¿recurrimos a api?

In [148]:
df.parque.value_counts()

El Cerrillo                   11
El Retiro                      9
Cerro Peñabel                  4
Tierno Galván                  4
Parque de los Estados          3
                              ..
Parque de Plata y Castañar     1
Martin Luther King             1
Planetario                     1
Parque Lineal                  1
Parque Martin Luther King      1
Name: parque, Length: 164, dtype: int64

In [149]:
df.parque.isnull().sum()

150

Tenemos 150 datos no válidos.

IDEM CON PARQUES. aquí nos da el nombre del parque más cercano. A lo mejor sería buen idea encontrar distancia al más cercano. ¿recurrimos a api?

In [150]:
df.hospital.value_counts()

General de Villalba                                 11
Hospital Universitario 12 de Octubre                10
Universitario 12 de Octubre                          7
Gregorio Marañón                                     6
Hospital General Universitario Gregorio Marañón      5
                                                    ..
Venerable Orden Tercera de San Francisco de Asís     1
Hospital Universitario 12 de Octubre                 1
Hospital Sanitas La Moraleja                         1
Fundación Hospital Alcorcón                          1
Centro de Salud Madrid Fuencarral                    1
Name: hospital, Length: 130, dtype: int64

In [151]:
df.hospital.isnull().sum()

162

Tenemos 162 datos no válidos.

IDEM CON HOSPITALES. aquí nos da el nombre del hospital más cercano. A lo mejor sería buen idea encontrar distancia al más cercano. ¿recurrimos a api?

In [152]:
df.gimnasio.value_counts()

Cross Fit Colmenar Viejo           11
XFitness La Elipa                   4
Go Fit                              4
VivaGym Gimnasio                    4
Holiday Gym Albufera                4
                                   ..
FitUp                               1
GO fit                              1
Holiday Gym Vergara                 1
Gimnasio AltaFit Cuatro Caminos     1
Gimnasio Victoria                   1
Name: gimnasio, Length: 160, dtype: int64

In [153]:
df.gimnasio.isnull().sum()

162

Tenemos 162 datos no válidos.

IDEM CON GIMNASIOS. aquí nos da el nombre del GImanasio más cercano. A lo mejor sería buen idea encontrar distancia al más cercano. ¿recurrimos a api?

➕➕ ¿buscar la información de supermercados cercanos podría ser interesante?

In [154]:
df.columns

Index(['rk', 'latitud', 'longitud', 'codigo_postal', 'pais', 'provincia',
       'ciudad', 'distrito', 'barrio', 'calle', 'planta', 'puerta', 'metro',
       'cercanias', 'centro_comercial', 'escuela', 'parque', 'hospital',
       'gimnasio'],
      dtype='object')

In [90]:
df.isna().sum()

rk                    0
latitud               2
longitud              2
codigo_postal         0
pais                  0
provincia             0
ciudad                0
distrito             98
barrio              357
calle                 0
planta                0
puerta               14
metro               154
cercanias           361
centro_comercial    166
escuela             161
parque              150
hospital            162
gimnasio            162
dtype: int64

# Características.csv

Contiene la información de las características de cada inmueble, aquí la rk es su identificador que esta compuesta por el UV (identificador de goelocalizacion) y por su estado (1 = sin reformar y 2 = reformado). Se divide pues podemos tener el mismo pisos con diferentes características según su estado.M
  

PREGUNTA: lo de que exista un piso en la BD reformado y sin reformar ¿es porque hay pisos en la mima planta, con las mismas características?

In [91]:
import pandas as  pd

In [92]:
df=pd.read_csv('../data/caracteristicas.csv',sep=';')

In [93]:
df.shape

(399, 27)

In [94]:
df.columns

Index(['rk', 'uv', 'estado', 'habitaciones', 'banos', 'metros', 'metrica',
       'fecha_construccion', 'orientacion', 'situacion',
       'certificado_energetico', 'portal', 'ascensor', 'parque_infantil',
       'terraza', 'trastero', 'piscina_comunitaria', 'cuarto_de_basura',
       'zonas_comunes', 'piscina', 'garaje', 'tejado', 'calefaccion',
       'certificaciones', 'jardin', 'aire_acondicionado', 'seguro_hogar'],
      dtype='object')

In [95]:
df.dtypes

rk                         object
uv                         object
estado                     object
habitaciones                int64
banos                       int64
metros                      int64
metrica                    object
fecha_construccion        float64
orientacion                object
situacion                  object
certificado_energetico     object
portal                      int64
ascensor                    int64
parque_infantil             int64
terraza                     int64
trastero                    int64
piscina_comunitaria         int64
cuarto_de_basura            int64
zonas_comunes               int64
piscina                     int64
garaje                      int64
tejado                    float64
calefaccion                 int64
certificaciones           float64
jardin                      int64
aire_acondicionado          int64
seguro_hogar                int64
dtype: object

In [96]:
df.isna().sum()

rk                        0
uv                        0
estado                    0
habitaciones              0
banos                     0
metros                    0
metrica                   0
fecha_construccion        1
orientacion               0
situacion                 7
certificado_energetico    2
portal                    0
ascensor                  0
parque_infantil           0
terraza                   0
trastero                  0
piscina_comunitaria       0
cuarto_de_basura          0
zonas_comunes             0
piscina                   0
garaje                    0
tejado                    1
calefaccion               0
certificaciones           5
jardin                    0
aire_acondicionado        0
seguro_hogar              0
dtype: int64

En general tenemos todos los datos casi completaos. Nos fallan:
    
    · Fecha de construcción: sólo uno no tiene el dato.
    · Situación: int/Ext: solo 7 no teemos el dato.
    · Certificado: solo 2 no tenemos el dato.
    · Tejado: Sólo uno no tenemos el dato

In [97]:
df.fecha_construccion.unique()

array([1963., 1960., 1966., 1972., 1983., 1980., 1977., 1971., 1969.,
       1900., 1987., 1946., 1964., 1974., 1984.,    0., 1965., 1999.,
       2007., 1918., 1981., 2000., 1944., 1955., 1991., 2008., 1982.,
       1976., 2003., 1970., 1967., 1952., 1950., 1920., 1956., 1986.,
       1940., 1978., 1905., 2004., 1998., 1988., 1962., 1910., 1973.,
       1975., 1996., 2011., 1992., 1954., 1953., 1935., 1985., 1930.,
       1915., 1949., 1934., 1870., 1898., 1929., 1958., 1957., 1925.,
       1884., 1961., 1928., 1933., 1959., 1968., 2005., 1994., 2002.,
       1936., 1979., 1990., 1923., 1911., 2001., 1951., 1914., 2066.,
       2010., 2015., 2019., 1890., 2006., 1922., 2009., 1945., 1861.,
         nan, 1941.])

Pocos datos incosistentes. A lo mejor nos convendría agrupar inmuebles más antiguos en una categoría ya que parece que entre 1936 y 1861 son simplemente viviendas muy antiguas.

In [98]:
df.orientacion.unique()

array(['Norte', 'Oeste', 'Suroeste', 'Sur', 'Este', 'Noroeste', 'Sureste',
       'Noreste', 'este', 'oeste', 'norte'], dtype=object)

Habría que unificar esta columna. Quizás sería buena idea dividir en dos columnas. Una N/S y otra E/O ya que podemos decir que las viviendas tiene claramente más o menos luz ( y por lo tanto su demanda) si suorientación es Norte o sur (ojo hemisferios) y 
que el parámetro E/O determina lo que busca el cliente en funció de si prefiere un tipo de luz u otro.

In [99]:

df.situacion.value_counts()

Exterior            323
Interior             57
Patio de manzana     12
Name: situacion, dtype: int64

Correcta esta columna

In [100]:
df.certificado_energetico.value_counts()

F            394
182000004      1
182000000      1
182000002      1
Name: certificado_energetico, dtype: int64

In [101]:
df.certificaciones.value_counts()

0.0    394
Name: certificaciones, dtype: int64

No entiendo mucho de certificados. Hasta donde se van por letas. No sé si los valores tipo 182000002 valores correctos o no. no sé a qué se refiere con "certificaciones"

In [102]:
df.portal.value_counts()

0    238
1    161
Name: portal, dtype: int64

In [103]:
df.ascensor.value_counts()

1    211
0    188
Name: ascensor, dtype: int64

In [104]:
df.parque_infantil.value_counts()

0    389
1     10
Name: parque_infantil, dtype: int64

In [105]:
df.terraza.value_counts()

0    282
1    117
Name: terraza, dtype: int64

In [106]:
df.trastero.value_counts()

0    342
1     57
Name: trastero, dtype: int64

In [107]:
df.piscina_comunitaria.value_counts()

0    377
1     22
Name: piscina_comunitaria, dtype: int64

In [108]:
df.piscina.value_counts()

0    393
1      6
Name: piscina, dtype: int64

In [109]:
(df.piscina.isna() != df.piscina_comunitaria.isna()).value_counts()

False    399
dtype: int64

Parece que no hay ninguna piscina distinta de piscina comunitaria, así que, a no ser que nos digan lo contrario parece que son casi redundantes. A no ser que consideremos piscinas privadas.

In [110]:
df.cuarto_de_basura.value_counts()

0    398
1      1
Name: cuarto_de_basura, dtype: int64

In [111]:
df.zonas_comunes.value_counts()

0    356
1     43
Name: zonas_comunes, dtype: int64

In [112]:
df.garaje.value_counts()

0    330
1     69
Name: garaje, dtype: int64

In [113]:
df.calefaccion.value_counts()

0    394
1      5
Name: calefaccion, dtype: int64

In [114]:
df.jardin.value_counts()

0    399
Name: jardin, dtype: int64

In [115]:
df.aire_acondicionado.value_counts()

0    391
1      8
Name: aire_acondicionado, dtype: int64

In [116]:
df.seguro_hogar.value_counts()

0    333
1     66
Name: seguro_hogar, dtype: int64

Parece ok estas columnas

In [117]:
df.tejado.value_counts()

0.0    398
Name: tejado, dtype: int64

No sé qué quiere decir la columna "tejado"

# Web.csv

Aquí tenemos el UV y estado con el precio al que lo comercializamos, para usar esa información de cara al presupuesto del cliente.

In [118]:
df=pd.read_csv('../data/web.csv',sep=';')


In [119]:
df.shape

(190, 8)

In [120]:
df.columns

Index(['uv', 'estado', 'tipo_vivienda', 'precio', 'precio_antiguo',
       'descuento', 'moneda', 'gastos_comunitarios'],
      dtype='object')

In [121]:
df.isna().sum()

uv                      0
estado                  0
tipo_vivienda           0
precio                  6
precio_antiguo         16
descuento              16
moneda                  0
gastos_comunitarios     0
dtype: int64

Tenemos algunos Nan en precio antiguo y descuento.

In [122]:
df.dtypes

uv                      object
estado                  object
tipo_vivienda           object
precio                 float64
precio_antiguo         float64
descuento              float64
moneda                  object
gastos_comunitarios    float64
dtype: object

In [123]:
df.uv.value_counts()

UV/2021/000457    4
UV/2021/000452    4
UV/2021/000447    4
UV/2021/000455    4
UV/2020/000310    2
                 ..
UV/2020/000213    1
UV/2021/000368    1
UV/2021/000357    1
UV/2020/000307    1
UV/2021/000388    1
Name: uv, Length: 177, dtype: int64

In [124]:
df.estado.value_counts()

1          161
2           28
Vendido      1
Name: estado, dtype: int64

In [125]:
df.tipo_vivienda.value_counts()

Piso       170
Casa         8
Local        5
Moradia      4
Oficina      1
Dúplex       1
Bajo         1
Name: tipo_vivienda, dtype: int64

In [126]:
df.precio.value_counts()

320000.0    6
180000.0    5
360000.0    5
340000.0    4
110000.0    4
           ..
97000.0     1
275000.0    1
199000.0    1
23000.0     1
460000.0    1
Name: precio, Length: 101, dtype: int64

Entiendo que el precio de la vivienda clikalia tiene algoritmos más que de sobra para estimarlo. ¿Podríamos trabajar en eso?

In [127]:
df.precio_antiguo.value_counts()

0.0         148
195000.0      2
299000.0      1
235000.0      1
90000.0       1
160000.0      1
149000.0      1
120000.0      1
85000.0       1
370000.0      1
130000.0      1
359000.0      1
350000.0      1
400000.0      1
360000.0      1
55000.0       1
275000.0      1
105000.0      1
550000.0      1
119000.0      1
204999.0      1
75000.0       1
329000.0      1
158000.0      1
190000.0      1
170000.0      1
Name: precio_antiguo, dtype: int64

In [128]:
df.descuento.value_counts()

0.0     149
5.0       3
9.0       3
4.0       3
6.0       3
2.0       2
11.0      2
13.0      2
12.0      2
10.0      1
16.0      1
8.0       1
7.0       1
3.0       1
Name: descuento, dtype: int64

In [129]:
(df.precio_antiguo.isna() != df.descuento.isna()).value_counts()

False    190
dtype: int64

Parece que siempre que tenemos precio antiguo tenemos el descuento que hemos aplicado

In [130]:
df.moneda.value_counts()

EURO    190
Name: moneda, dtype: int64

In [131]:
df.gastos_comunitarios.value_counts()

0.00      145
12.00       8
150.00      4
1.00        4
30.00       3
140.00      2
40.00       2
183.00      1
153.77      1
60.00       1
30.47       1
115.00      1
219.51      1
117.00      1
33.00       1
64.49       1
18.00       1
62.51       1
64.82       1
59.72       1
200.00      1
13.33       1
49.00       1
134.04      1
50.17       1
89.41       1
157.00      1
70.00       1
84.97       1
Name: gastos_comunitarios, dtype: int64